### PR TITLE
run_status/failure_sensor monitor_all_repositories -> code_locations 1/2

### PR DESCRIPTION
**Pull request by yuhan**
## Summary & Motivation
this pr adds a `monitor_all_code_locations` param and deprecates `monitor_all_repositories` in `run_status_sensor` and `run_failure_sensor`

this comes up in a couple of support answers and it felt awkward to me that the code still talks about "repositories" when lots of users have moved past it.

side question: should we wait to deprecate in 1.7? imo it doesn't matter much cos this is more of fixing a long-standing confusion.

## How I Tested These Changes
- [x] add test to test code locations

### CODE REVIEW

**Comment by yuhan** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #19555** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/19555?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/19555?utm_source=stack-comment).

**Comment by yuhan** [comment]
converting back to draft as the current tests don't actually capture code locations

File: python_modules/dagster/dagster/_core/definitions/run_status_sensor_definition.py [hunk 1]
```diff
@@ -505,14 +526,15 @@ def inner(
             sensor_name = name
 
         jobs = monitored_jobs if monitored_jobs else job_selection
+        monitor_all = monitor_all_code_locations or monitor_all_repositories
```
Comment (OwenKephart) [inline python_modules/dagster/dagster/_core/definitions/run_status_sensor_definition.py:529]: nit: we have this method: https://sourcegraph.com/github.com/dagster-io/dagster@143d555c5b25c63ee690632292f17e9122bc92ec/-/blob/python_modules/dagster/dagster/_utils/warnings.py?L18 which handles some edge cases automatically (e.g. what if user supplies both)

Comment (yuhan) [reply python_modules/dagster/dagster/_core/definitions/run_status_sensor_definition.py:529]: not nit actually - i was being lazy but ty for capturing it!

Comment (yuhan) [reply python_modules/dagster/dagster/_core/definitions/run_status_sensor_definition.py]: in order to capture the case where user supplies both, i converted the func signatures to accept explicitly nullable params (previously, it defaulted to False so we couldn't error here)

**Review by OwenKephart — APPROVED** [review]

**Review by jamiedemaria — APPROVED** [review]

**Comment by yuhan** [comment]
### Merge activity

* **Feb 6, 1:17 PM PST**: @yuhan started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/19555).

**Pull request merged**